contentcuration/contentcuration/views/users.py

```diff
@@ -97,6 +97,23 @@ def send_invitation_email(request):
     return Response(InvitationSerializer(invitation).data)
 
 
+@api_view(['GET'])
+@authentication_classes((SessionAuthentication,))
+@permission_classes((IsAuthenticated,))
+def deferred_user_data(request):
+    if request.GET.get("settings"):
```

Comment:
Could you add some detail about why we have the two different paths here? I notice we're calculating `available_space` in one code path, and just getting the data from a `user` method in another, and it wasn't clear to me why we have these two paths and if we can be sure they will never diverge in terms of how `available_space` gets calculated.

Reply:
Yeah - I could just add two separate endpoints too, but was being slightly lazy.

I am not 100% confident that the two would not diverge - as I can imagine a situation where the same file is reused by multiple kinds, so I can imagine that being a problem?

The reason for doing this was that doing the calculations twice was twice as expensive, unsurprisingly - taking up to 6 seconds on my dev machine.

Reply:
Is that with a large number of files, or for a fairly standard use case? Also, which case will get 'hit' more - the `settings` variant or the non-settings one?

Reply:
The non-settings case should get hit more often. Have simplified it so that the data is always consistent.